modules/frontend/tracebyidsharding.go

```diff
@@ -259,14 +260,25 @@ func createBlockBoundaries(queryShards int) [][]byte {
 	for i := 0; i < queryShards+1; i++ {
 		blockBoundaries[i] = make([]byte, 16)
 	}
-	const MaxUint = uint64(^uint8(0))
+
+	// bucketSz is the min size for the bucket
+	bucketSz := (math.MaxUint64 / uint64(queryShards))
+	// numLarger is the number of buckets that have to be bumped by 1
+	numLarger := (math.MaxUint64 % uint64(queryShards))
+	boundary := uint64(0)
 	for i := 0; i < queryShards; i++ {
-		binary.LittleEndian.PutUint64(blockBoundaries[i][:8], (MaxUint/uint64(queryShards))*uint64(i))
-		binary.LittleEndian.PutUint64(blockBoundaries[i][8:], 0)
+		binary.BigEndian.PutUint64(blockBoundaries[i][:8], boundary)
+		binary.BigEndian.PutUint64(blockBoundaries[i][8:], 0)
+
+		boundary += bucketSz
+		if numLarger != 0 {
+			numLarger--
+			boundary++
```

Comment:
👍 I like this way of distributing the extras.

Reply:
i stole this from stack overflow :)